pkg/discovery/dtofactory/group_dto_builder.go

```diff
@@ -63,77 +70,146 @@ func (builder *groupDTOBuilder) BuildGroupDTOs() []*proto.GroupDTO {
 			continue
 		}
 
-		// Groups for the Pod and containers belonging to the group
-		for etype, memberList := range entityGroup.Members {
+		// create groups for each configured entity type
+		for etype, resizeGroup := range GROUP_RESIZE_FLAG_MAP {
+			groupDTOs := builder.createGroupsByEntityType(entityGroup, etype, resizeGroup)
+
+			result = append(result, groupDTOs...)
+		}
+	}
+
+	return result
+}
+
+func (builder *groupDTOBuilder) createGroupsByEntityType(entityGroup *repository.EntityGroup,
+	entityType metrics.DiscoveredEntityType, resizeGroup bool) []*proto.GroupDTO {
 
-			// group id created using the parent type, name and target identifier
-			id := fmt.Sprintf("%s-%s[%s]", entityGroup.GroupId, builder.targetId, etype)
-			displayName := fmt.Sprintf("%ss By %s [%s]", etype, entityGroup.GroupId, builder.targetId)
+	var result []*proto.GroupDTO
 
-			var protoType proto.EntityDTO_EntityType
-			protoType, foundType := ENTITY_TYPE_MAP[etype]
-			if !foundType {
-				glog.Errorf("Invalid member entity type %s", etype)
-				continue
-			}
+	// member type
+	var protoType proto.EntityDTO_EntityType
+	protoType, foundType := ENTITY_TYPE_MAP[entityType]
+	if !foundType {
+		glog.Errorf("Invalid member entity type %s", entityType)
+		return []*proto.GroupDTO{}
+	}
 
-			// static group
-			groupBuilder := group.StaticGroup(id).
-				OfType(protoType).
-				WithEntities(memberList).
-				WithDisplayName(displayName)
+	// member list
+	memberList, etypeExists := entityGroup.Members[entityType]
 
-			// build group
-			groupDTO, err := groupBuilder.Build()
-			if err != nil {
-				glog.Errorf("Error creating group dto  %s::%s", id, err)
-				continue
-			}
+	if !etypeExists {
+		return []*proto.GroupDTO{}
+	}
 
-			result = append(result, groupDTO)
+	// group id for parent group
+	groupId := fmt.Sprintf("%s", entityGroup.GroupId)
 
-			glog.V(4).Infof("groupDTO  : %++v", groupDTO)
-		}
+	// resize setting for container groups based on the parent type of the group
+	// and if sub groups will be created.
+	// If sub groups are created, the resize setting is applied on the sub-groups
+	resizeFlag := resizeGroup && consistentResize(entityGroup) && resizeParentGroup(entityGroup, entityType)
```

Comment:
I see multiple places that we are checking if it's a container group or not, like GROUP_RESIZE_FLAG_MAP and in resizeParentGroup, can we merge the logic?